### PR TITLE
Changing the generation of fake template with a decaying exponential and fix torch locally exclusive

### PR DESCRIPTION
**Pull request by yger**
In this PR, we change the way fake templates are generated. Instead of a powerlaw for the decay of the amplitudes as function of the distances, we use a decaying exponential with a spatial_constant. This seems to be more biophysically plausible, and it introduces less artefacts

### CODE REVIEW

**Comment by alejoe91** [comment]
@yger can you fix the failing tests

**Comment by yger** [comment]
Yes, I will, but I have no idea why my modification in the fake templates is triggering a negative std for the template. This make no sense, I'll have a look

**Comment by alejoe91** [comment]
Can you try with this? https://github.com/SpikeInterface/spikeinterface/pull/2651

**Comment by yger** [comment]
found it, it was some default alpha range. Tests should pass now

**Comment by yger** [comment]
Looks like this is triggering an error in a peak detection function with torch... @samuelgarcia , did you made this functio?

**Comment by alejoe91** [comment]
I ported it, I'll take a look

**Comment by alejoe91** [comment]
@yger should be fixed!

Plus it fixes #2213

File: src/spikeinterface/core/tests/test_analyzer_extension_core.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def test_ComputeWaveforms(format, sparse):
 def test_ComputeTemplates(format, sparse):
     sorting_analyzer = get_sorting_analyzer(format=format, sparse=sparse)
 
-    sorting_analyzer.compute("random_spikes", max_spikes_per_unit=20, seed=2205)
+    sorting_analyzer.compute("random_spikes", max_spikes_per_unit=500, seed=2205)
```
Comment (samuelgarcia) [inline src/spikeinterface/core/tests/test_analyzer_extension_core.py:114]: This slow down a lot tests. I prefer a small number!!

Comment (alejoe91) [reply src/spikeinterface/core/tests/test_analyzer_extension_core.py:114]: It forsnt

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**